internal/services/azurestackhci/stack_hci_storage_path_data_source.go

```diff
@@ -0,0 +1,120 @@
+package azurestackhci
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/storagecontainers"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/extendedlocation/2021-08-15/customlocations"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type StackHCIStoragePathDataSource struct{}
+
+var _ sdk.DataSource = StackHCIStoragePathDataSource{}
+
+type StackHCIStoragePathDataSourceModel struct {
+	Name              string                 `tfschema:"name"`
+	ResourceGroupName string                 `tfschema:"resource_group_name"`
+	Location          string                 `tfschema:"location"`
+	CustomLocationId  string                 `tfschema:"custom_location_id"`
+	Path              string                 `tfschema:"path"`
+	Tags              map[string]interface{} `tfschema:"tags"`
+}
+
+func (r StackHCIStoragePathDataSource) ResourceType() string {
+	return "azurerm_stack_hci_storage_path"
+}
+
+func (r StackHCIStoragePathDataSource) ModelObject() interface{} {
+	return &StackHCIStoragePathDataSourceModel{}
+}
+
+func (r StackHCIStoragePathDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9][\-\.\_a-zA-Z0-9]{1,78}[a-zA-Z0-9]$`),
+				"name must begin and end with an alphanumeric character, be between 3 and 80 characters in length and can only contain alphanumeric characters, hyphens, periods or underscores.",
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
```

Comment:
We have a specific resource group common schema for data sources that we should use instead

```suggestion
		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
```

Reply:
fixed